### ISSUE TITLE
Zx Microdrives files support (.mdr)

### ISSUE THREAD

**Issue by PakkunKinoppi** [raydac/zxpoly#68]
Hello, I am wondering if you could consider implementing Interface One as a selectable Add-on?

This ROM I meant: if1-2.rom - Interface 1 v2 ROM
https://github.com/rastersoft/fbzx/blob/master/data/spectrum-roms/if1-2.rom

Thank you
![sinclair_zx_interface_1-detail](https://user-images.githubusercontent.com/52820759/162692726-7f869099-07a5-4e0e-a411-0fc136eb93af.png)

**Comment by Alessandro1970** [comment]
Is this ask to use mdr (microdrive) image file ?

[tape1.zip](https://github.com/raydac/zxpoly/files/8463433/tape1.zip)

Or anything else ?

**Comment by Alvarez81** [comment]
> _A wedge-shaped device fitting underneath the ZX Spectrum, the ZX Interface 1 contained 8 kB of [ROM](https://en.wikipedia.org/wiki/Read-only_memory) comprising the control software for the Microdrives, an RS-232 port and network interface. This extended the error handler in the [Sinclair BASIC](https://en.wikipedia.org/wiki/Sinclair_BASIC) to allow extra keywords to be used_

This is the first and the more used interface after kempston joy. Fuse has got it but how about RS232 support ?

**Comment by PakkunKinoppi** [comment]
Yes, ZX1 Interface adds some basic commands to load/save programs on microdrives (.mdr files). Thank you.

**Comment by PakkunKinoppi** [comment]
These are microdrive file images:
1) https://spectrumcomputing.co.uk/pub/sinclair/utils/m/MicrodriveUtilities.mdr.zip
2) https://spectrumcomputing.co.uk/pub/sinclair/compilations/b/BusinessCartridge.mdr.zip
3) https://spectrumcomputing.co.uk/pub/sinclair/compilations/d/DemoCartridge.mdr.zip
4) https://spectrumcomputing.co.uk/pub/sinclair/compilations/g/GamesCartridge(fixed).mdr.zip
5) https://spectrumcomputing.co.uk/zxdb/sinclair/entries/0036709/MicrodriveBackupUtility.mdr.zip
6) https://spectrumcomputing.co.uk/pub/sinclair/utils/m/MicrodriveUtilitySystem.mdr.zip
7) https://spectrumcomputing.co.uk/zxdb/sinclair/entries/0017602/RAMDOS.mdr.zip

**Comment by Alvarez81** [comment]
In preference panel a new selection path loader for locate and load the IF1-1.rom and IF1-2.rom as like as for zx48/128k would be better.

**Comment by Haroldo64** [comment]
In any case, I thank you very much for this very useful and necessary enhancement to be able to recover many software from the web.

**Comment by Alvarez81** [comment]
Other *.mdr (microdrive) image files:

https://tzxvault.org/microdrv.htm

**Comment by Alvarez81** [comment]
Is there any way to convert *.mdr files to *.TAP or something usefull for ZxPoly ?

**Comment by Haroldo64** [comment]
@Alvarez81  ...keep calm, @raydac will support mdr files and some of ZX Interface 1.

BTW I think that RS232 support in not so important at this time for an emulator...
